Documentation/ApiOverview/ContentElements/CustomBackendPreview.rst

```diff
@@ -14,6 +14,72 @@ following sections describe how to achieve that.
 A preview renderer is used to facilitate (record) previews in TYPO3. This
 class is responsible for generating the preview and the wrapping.
 
+The default preview renderer is :php:`StandardContentPreviewRenderer` and handles
+the Core's built-in content types (field :sql:`CType` in table :sql:`tt_content`).
+
+Extend the default PreviewRenderer
+==================================
+
+There are two ways of how to provide previews for your own custom content types:
+via page TSconfig or event listener.
+
+page TSconfig
+----------------
+
+..  code-block:: typoscript
+
+    mod.web_layout {
+      tt_content {
+        preview {
+          # Your CType
+          example_ctype = EXT:my_extension/Resources/Private/Templates/Preview/ExampleCType.html
+        }
+      }
+    }
+
+For more details see :ref:`TSconfig Reference <t3tsconfig:pageweblayoutpreview>`.
+
+Event listener
+--------------
+
+..  versionadded:: 12.0
+    Since version 12.0 this technique replaces the former hook
+    :php:`$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['cms/layout/class.tx_cms_layout.php']['tt_content_drawItem']`
+
+The event :php:`PageContentPreviewRenderingEvent` is being dispatched by the
```

Comment:
```suggestion
The event :ref:`PageContentPreviewRenderingEvent` is being dispatched by the
```

So the event is linked to the appropriate page:
https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Events/Events/Backend/PageContentPreviewRenderingEvent.html

I also want to see the example on the event page. If someone searches "Big G" for the event, possibilities are high, they are getting the event page as first result. You can then omit the example here. I suggest to add the example in a separate PR.

For an example, have a look into this redirect event:
https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Events/Events/Redirects/RedirectWasHitEvent.html
The snippets are embedded from external files, so we can check the syntax and migrate PHP code via Rector to a newer PHP version when necessary.

Reply:
I created a PR for the event example: #2814 . So best way would be to reference this event and remove the example from here. What do you think?

Reply:
@brotkrueml, sure, like that idea! But isn't that code example of yours missing `services:` and most importantly the `event` property?

Reply:
The event property is not necessary since v11. Reflection is doing the job. And we leave out all the introductionary things in Services.yaml in event overview, just the relevant part.

Reply:
@brotkrueml, done. Please have a look.